### PR TITLE
 7926 & 7250 finish ARP data filter

### PR DESCRIPTION
**Pull request by brett-varney**
**High level description:**

update overview bar and info boxes to use data filtered by selected law(s)

**Technical details:**

I accidentally updated the overview while working on the info boxes, so combined the tickets into this. Mostly just replaced "defCodes" with "defcParams"

**JIRA Tickets:**
[DEV-7250](https://federal-spending-transparency.atlassian.net/browse/DEV-7250)
[DEV-7926](https://federal-spending-transparency.atlassian.net/browse/DEV-7926)

**Mockup:**
na

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [x] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [na] Verified cross-browser compatibility: Chrome, Safari, Firefox, Internet Explorer, Edge
- [na] Verified mobile/tablet/desktop/monitor responsiveness
- [na] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [na] Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)
- [na] [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
- [na] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [x] Design review complete `if applicable`
- [na] [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [x] Code review complete

### CODE REVIEW

File: src/js/models/v2/covid19/BaseOverview.js [hunk 1]
```diff
@@ -10,11 +10,11 @@ const BaseOverview = {
         data.funding.forEach((code) => {
             this[`_defCode_${code.def_code}_funding`] = code.amount;
         });
-        this._totalBudgetAuthority = data.total_budget_authority || null;
+        this._totalBudgetAuthority = data.total_budget_authority || null + (data.additional ? data.additional.total_budget_authority : 0);
```
Comment (ebdabbs) [inline src/js/models/v2/covid19/BaseOverview.js:13]: This logic should apply _only_ to the bar chart visualization, not the summary insights throughout the page. I believe we're currently referencing these 3 fields for both the chart and the summary insights. We could create 3 new fields to be used by the bar chart.

File: tests/containers/covid19/Covid19Container-test.jsx [hunk 2]
```diff
@@ -10,8 +10,8 @@ import * as apis from 'apis/disaster';
 import * as actions from 'redux/actions/covid19/covid19Actions';
 import { useQueryParams } from 'helpers/queryParams';
 import Covid19Container from 'containers/covid19/Covid19Container';
-import { mockDefCodes } from '../../mockData/helpers/disasterHelper';
-import { defaultState } from '../../testResources/defaultReduxFilters';
+import { mockDefCodes, mockDefcParams } from '../../mockData/helpers/disasterHelper';
+// import { defaultState, covid19DefaultState } from '../../testResources/defaultReduxFilters';
```
Comment (ebdabbs) [inline tests/containers/covid19/Covid19Container-test.jsx:14]: Can this line be removed instead of commented?

**Review by ebdabbs — CHANGES_REQUESTED** [review]
I'd also like to add a loading message for the bar chart when the request is in flight as you switch the filter. I realize this isn't in the AC for DEV-7250, so up to you whether you'd like to add it here or create a follow up ticket!

File: tests/containers/covid19/Covid19Container-test.jsx (hunk 2 above)
Comment (brett-varney) [reply tests/containers/covid19/Covid19Container-test.jsx:14]: yes, sorry, left over from trying to figure a problem out

File: src/js/models/v2/covid19/BaseOverview.js (hunk 1 above)
Comment (brett-varney) [reply src/js/models/v2/covid19/BaseOverview.js:13]: gotcha

**Comment by brett-varney** [comment]
> 
> 
> I'd also like to add a loading message for the bar chart when the request is in flight as you switch the filter. I realize this isn't in the AC for DEV-7250, so up to you whether you'd like to add it here or create a follow up ticket!
> 

I am combing through all the layers of the page, and I cannot figure out how to do this.

**Review by ebdabbs — CHANGES_REQUESTED** [review]
- The budget authority number that we use to create the d3 linear scale should be updated to reference `_totalBudgetAuthorityForBar` - [this line](https://github.com/fedspendingtransparency/usaspending-website/blob/qat/src/js/components/covid19/amountsVisualization/AmountsVisualization.jsx#L46). Without that, the chart can get cut off if `_totalBudgetAuthorityForBar` and `_totalBudgetAuthority` differ
- The number that we use to calculate the tooltip percentage also needs to be updated to reference `_totalBudgetAuthorityForBar` - [this line](https://github.com/fedspendingtransparency/usaspending-website/blob/qat/src/js/components/covid19/amountsVisualization/AmountsVisualization.jsx#L92)
- I created [DEV-7947](https://federal-spending-transparency.atlassian.net/browse/DEV-7947) for the loading state and added a testing note to DEV-7250

**Comment by brett-varney** [comment]
> 
> 
>     * The budget authority number that we use to create the d3 linear scale should be updated to reference `_totalBudgetAuthorityForBar` - [this line](https://github.com/fedspendingtransparency/usaspending-website/blob/qat/src/js/components/covid19/amountsVisualization/AmountsVisualization.jsx#L46). Without that, the chart can get cut off if `_totalBudgetAuthorityForBar` and `_totalBudgetAuthority` differ
> 
>     * The number that we use to calculate the tooltip percentage also needs to be updated to reference `_totalBudgetAuthorityForBar` - [this line](https://github.com/fedspendingtransparency/usaspending-website/blob/qat/src/js/components/covid19/amountsVisualization/AmountsVisualization.jsx#L92)
> 
>     * I created [DEV-7947](https://federal-spending-transparency.atlassian.net/browse/DEV-7947) for the loading state and added a testing note to DEV-7250
> 
>     * Still missing unit tests for the new fields added to the `BaseOverview` data model ([this file](https://github.com/fedspendingtransparency/usaspending-website/blob/qat/tests/models/covid19/BaseOverview-test.js))

Thanks for all this, getting to them

**Review by ebdabbs — CHANGES_REQUESTED** [review]
I realized we also need to update `tooltipMapping` in `dataMapping/covid19/amountsVisualization.js` so that the numbers in the tooltip also use the "forBar" fields!

**Review by ebdabbs — COMMENTED** [review]
Now I am getting a console error and a blank screen when I hover over the bar chart
```
Uncaught TypeError: Cannot read properties of undefined (reading 'title')
    at tooltipData (AmountsVisualization.jsx:129)
```

**Review by ebdabbs — APPROVED** [review]

**Pull request merged**